packages/interface/src/interface.ts

```diff
@@ -0,0 +1,165 @@
+ *
+ * If ShadyCSS is active, then after styleElement is called on the first
+ * instance of this element, <style> tags within this template will be moved to
+ * the <head> and re-written to use globally scoped rules that emulate scoped
+ * behavior.
+ *
+ * Note that LitElement and Polymer Library users do not typically need to call
+ * this function, because it is called automatically.
+ */
+export function prepareTemplate(
+  template: HTMLTemplateElement,
+  customElementName: string
+): void {
+  if (scopedStylesArePolyfilled()) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    polyfillsWindow.ShadyCSS!.prepareTemplate(template, customElementName);
+  }
+}
+
+/**
+ * Activate scoped styles for the given element instance. This function should
+ * typically be called inside connectedCallback. The template of this element
+ * class must already have been registered with prepareTemplate.
+ *
+ * If ShadyCSS is not active, then this function does nothing.
+ *
+ * Note that LitElement and Polymer Library users do not typically need to call
+ * this function, because it is called automatically.
+ */
+export function styleElement(element: HTMLElement): void {
+  if (scopedStylesArePolyfilled()) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    polyfillsWindow.ShadyCSS!.styleElement(element);
+  }
+}
+
+/**
+ * Propagate CSS custom properties on this element to all descendant shadow
+ * roots, and optionally set new ones.
+ *
+ * Uses ShadyCSS custom property emulation if the polyfill is active, otherwise
+ * calls native style.setProperty.
+ */
+export function styleSubtree(
+  element: HTMLElement,
+  properties?: CustomPropertyValues
+): void {
+  if (customPropertiesArePolyfilled()) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    polyfillsWindow.ShadyCSS!.styleSubtree(element, properties);
+  } else if (properties != null) {
+    setProperties(element, properties);
+  }
+}
+
+/**
+ * Set CSS custom properties on the document and propagate to all shadow roots.
+ *
+ * Uses ShadyCSS custom property emulation if that polyfill is active, otherwise
+ * uses native style.setProperty.
+ */
+export function styleDocument(properties: CustomPropertyValues): void {
+  if (customPropertiesArePolyfilled()) {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    polyfillsWindow.ShadyCSS!.styleDocument(properties);
+  } else if (properties != null) {
+    setProperties(document.body, properties);
+  }
+}
+
+/**
+ * Propagate custom properties from the given `document` level `<style>` tag.
+ * Changes occur asynchronously as a microtask.
+ *
+ * If ShadyCSS is not active, or if the `custom-style-interface` library has not
+ * been loaded, then this function does nothing.
+ *
+ * Note that the `custom-style-interface` library is not automatically loaded by
+ * the `loader` or `bundle` scripts. It must be loaded separately.
+ */
```

Comment:
Should we mention https://github.com/webcomponents/polyfills/tree/master/packages/shady-css-scoped-element?

Reply:
Yep, but that's not published yet (is on my near-term todo list).

On that topic, it seems like shady-css-scoped-element (and maybe this new interface module too?) should automatically work without the user having to load custom-style-interface themselves. custom-style-interface is pretty small, so I wonder if we should just bundle custom-style-interface with shadycss? Otherwise, it could be a dependency of shady-css-scoped-element, like it is in Polymer.